src/Support/HttpCacheMiddleware.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NiemandOnline\HeptaConnect\Portal\Amiibo\Support;
+
+use Heptacom\HeptaConnect\Portal\Base\Web\Http\Contract\HttpClientMiddlewareInterface;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+final class HttpCacheMiddleware implements HttpClientMiddlewareInterface
+{
+    private array $cache = [];
+
+    public function process(RequestInterface $request, ClientInterface $handler): ResponseInterface
+    {
+        $cacheKey = (string) $request->getUri();
```

Comment:
This should only happen for "read only" requests like GET and HEAD

Reply:
Done